packages/core/admin/_internal/node/vite/config.ts

```diff
@@ -0,0 +1,134 @@
+import type { BuildContext } from '../createBuildContext';
+import { getMonorepoAliases } from '../core/aliases';
+import { buildFilesPlugin, runtimeRewritePlugin } from './plugins';
+import findRoot from 'find-root';
+
+const resolveBaseConfig = async (ctx: BuildContext): Promise<InlineConfig> => {
+  const target = browserslistToEsbuild(ctx.target);
+
+  return {
+    root: ctx.cwd,
+    base: ctx.basePath,
+    build: {
+      emptyOutDir: false, // Rely on CLI to do this
+      outDir: ctx.distDir,
+      target,
+    },
+    cacheDir: 'node_modules/.strapi/vite',
+    configFile: false,
+    define: {
+      'process.env': ctx.env,
+    },
+    esbuild: {
+      loader: 'jsx',
+      include: /.*\.jsx?$/,
+      exclude: [],
+    },
+    envPrefix: 'STRAPI_ADMIN_',
+    optimizeDeps: {
+      include: [
+        // pre-bundle React dependencies to avoid React duplicates,
+        // even if React dependencies are not direct dependencies
+        // https://react.dev/warnings/invalid-hook-call-warning#duplicate-react
+        'react',
+        `react/jsx-runtime`,
+        `react/jsx-dev-runtime`,
+        'react-dom/client',
+      ],
+    },
+    resolve: {
+      // https://react.dev/warnings/invalid-hook-call-warning#duplicate-react
+      dedupe: ['react', 'react-dom'],
+    },
+    plugins: [react(), runtimeRewritePlugin(ctx), buildFilesPlugin(ctx)],
+  };
+};
+
+const resolveProductionConfig = async (ctx: BuildContext): Promise<InlineConfig> => {
+  const {
+    options: { minify, sourcemaps },
+  } = ctx;
+
+  const baseConfig = await resolveBaseConfig(ctx);
+
+  return {
+    ...baseConfig,
+    logLevel: 'silent',
+    mode: 'production',
+    build: {
+      ...baseConfig.build,
+      assetsDir: '',
+      minify: false,
+      sourcemap: true,
+      rollupOptions: {
+        input: {
+          strapi: ctx.entry,
+        },
+      },
+    },
+  };
+};
+
+const resolveDevelopmentConfig = async (ctx: BuildContext): Promise<InlineConfig> => {
+  const monorepo = await loadStrapiMonorepo(ctx.cwd);
+  const baseConfig = await resolveBaseConfig(ctx);
+
+  return {
+    ...baseConfig,
+    resolve: {
+      ...baseConfig.resolve,
+      alias: {
+        ...baseConfig.resolve?.alias,
+        // ...getMonorepoAliases({ monorepo }),
+      },
+    },
+    plugins: [
+      ...(baseConfig.plugins ?? []),
+      monorepo?.path
+        ? {
+            name: 'treat-js-files-as-jsx',
+            async transform(code, id) {
+              if (!id.match(/.*\.js$/)) {
+                return null;
+              }
+
+              // Use the exposed transform from vite, instead of directly
+              // transforming with esbuild
+              return transformWithEsbuild(code, id, {
+                loader: 'tsx',
+                jsx: 'automatic',
+              });
+            },
+          }
+        : null,
+    ],
+    server: {
+      middlewareMode: true,
+      open: ctx.options.open,
+    },
+    appType: 'custom',
```

Comment:
I thought setting this would indicate I handle routing so it would let `/admin/init` pass for example – apparently not...